src/client.rs

```diff
@@ -267,17 +378,36 @@ pub enum ClientFlowEvent {
         /// (e.g. [`Code::Alert`](imap_codec::imap_types::response::Code::Alert)).
         status: Status<'static>,
     },
+    AuthenticateStarted {
+        handle: ClientFlowCommandHandle,
+    },
+    /// Server is requesting (more) authentication data.
```

Comment:
The doc should mention that `authenticate_continue` should be called afterwards.